### DISCUSSION TITLE
Can you set up GPG signing in GitHub Desktop?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#48667] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Recently have I set up GPG signing for commits, which included setting it up here on GitHub and also configuring git to use it too.

However, when trying to use GitHub Desktop will it not accept my attempts at committing changes as it errors out with the following (Translated from german):
```
gpg: skipped "<key>": Not a secret key*
gpg: Signing failed: Not a secret key*
error: gpg failed to sign the data
fatal: failed to write commit object
```
*Note that the original says "Kein geheimer Schlüssel" which can mean the above but also "No secret key". I assume it's the top one. Also `<key>` obviously is that random string that is a public/private key.

I'm now struggling to find out if you can configure GitHub to accept GPG signinging and if so how. Google searches mostly gave me replies like "GitHub Desktop doesn't support GPG signing" which if true would be really embarassing to say the least.

Either way, I'm now asking if anyone knows if you can set up GitHub Desktop to use the GPG signing when doing commits, as right now my only workaround is doing a `git add .` followed by `git commit -S -m "Message"` in Git Bash followed by a push from GitHub Desktop because apparently I lack authorization for pushing to GitHub in Git Bash and don't want to set up Git for Auth with GitHub for a single case like that.

**Comment by DNin01** [comment]
It should sign your commits as long as Git is configured to sign commits by default. [This](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) should tell you how to do that. If that doesn't work, you might also want to try committing from Git to see if that's the problem (you might be able to use a local repository, that is, one in your computer's files), and if it is or you're unsure about what the problem is, try reading GitHub Docs → [Managing commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification) and make sure you're following all the steps.

**Reply by Andre601** [reply]
Git is configured on a Global level (Using `--global`) to sign commits, so the issue must either be on the repository (Which I doubt as other repos work fine when committing using Git Bash or another tool like IntelliJ's inbuild git feature) or actually with GitHub Desktop as committing through Git Bash itself is giving me the passphrase prompt and allows me to have the commits signed properly.

**Reply by DNin01** [reply]
Have you updated GitHub Desktop recently? I know older versions of GitHub Desktop don't support commit signing.

**Reply by Andre601** [reply]
I've updated GH Desktop, yet the same issue persists here.

**Reply by DNin01** [reply]
It still didn't work? I don't have any more ideas, then. Sorry about that.
